src/main/java/org/kainos/ea/db/AuthDao.java

```diff
@@ -0,0 +1,111 @@
+package org.kainos.ea.db;
+
+import org.apache.commons.lang3.time.DateUtils;
+import org.kainos.ea.cli.Credential;
+import org.kainos.ea.cli.UserRole;
+import org.kainos.ea.client.AuthenticationException;
+import org.kainos.ea.client.GenericActionFailedException;
+
+import java.sql.*;
+import java.util.Date;
+import java.util.UUID;
+
+public class AuthDao implements IAuthenticationSource {
+    @Override
+    public boolean validateLogin(Credential credential) {
+        try {
+            Connection conn = DatabaseConnector.getConnection();
+            String SQL = "SELECT Password FROM `User` WHERE Username = ?";
+            PreparedStatement st = conn.prepareStatement(SQL);
+
+            st.setString(1,credential.getUsername());
+
+            ResultSet rs = st.executeQuery();
+
+            while(rs.next()) {
+                return rs.getString("Password").equals(credential.getPassword());
+            }
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            return false;
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean registerUser(Credential credential) throws GenericActionFailedException, AuthenticationException {
+        try {
+            Connection conn = DatabaseConnector.getConnection();
+            System.out.println(credential.getUsername() + " " + credential.getPassword());
+            String SQL = "INSERT INTO User(Username,Password,RoleID) VALUES(?,?,2)";
```

Comment:
remove hardcoded 2

Reply:
We actually need this to default a user to a specific role. We might want to consider a change in the SQL to default this value so it doesn't need to be specified?

Reply:
maybe just add the hardcoded value in the setInt function to make it easy to update later.

Reply:
Yeah good shout, sorted